lib/src/scoring.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2017, Kevin Moore. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math' as math;
+
+class Summary {
+  final num max;
+  final num min;
+  final num mean;
+
+  Summary._(this.max, this.min, this.mean) {
+    assert(max >= mean);
+    assert(mean >= min);
+  }
+
+  factory Summary(Iterable<num> numbers) {
+    var total = 0.0;
+    num max, min;
+
+    var count = 0;
+    for (var i in numbers) {
+      //TODO(kevmoo): validate i for Nan/Infinity
+      count++;
+      total += i;
+
+      if (max == null || i > max) {
```

Comment:
right...
the if can be spared though with:
max = math.max(i, max ?? i);

but not that important